DNN Platform/DotNetNuke.DependencyInjection/Extensions/TypeExtensions.cs

```diff
@@ -26,10 +28,15 @@ public static class TypeExtensions
         /// <param name="assembly">
         /// The assembly to retrieve all types from.
         /// </param>
+        /// <param name="logger">
+        /// A optional <see cref="ILog"/> object. This will log any messages from the exception catching to the logs as an Error.
+        /// </param>
         /// <returns>
         /// An array of all <see cref="Type"/> in the given <see cref="Assembly"/>.
-        /// </returns>
-        public static Type[] SafeGetTypes(this Assembly assembly)
+        /// </returns>
+#pragma warning disable CS3001 // Argument type is not CLS-compliant
+        public static Type[] SafeGetTypes(this Assembly assembly, ILog logger = null)
+#pragma warning restore CS3001 // Argument type is not CLS-compliant

```

Comment:
@mtrutledge What was not CLS compliant here. Since this class is public, it would be nice to be able to fix it so it is CLS compliant ?

Reply:
Could it be that it is because the DotNetNuke.Instrumentation library is simply not declared as CLSCompliant ? What that the issue

Reply:
I could not find a reason. I will check the DotNetNuke.Instrumentation dll and see if it needs the CLSCompliant added. if it does I will add it and update the PR.

Reply:
Yeah, the DotNetNuke.Instrumentation library was missing the CLS Compliant attribute. Added it and updated the PR.